sdk/eventhub/event-hubs/src/receiver.ts

```diff
@@ -154,25 +183,26 @@ export class Receiver {
     maxWaitTimeInSeconds?: number,
     cancellationToken?: Aborter
   ): Promise<ReceivedEventData[]> {
-    const bReceiver = BatchingReceiver.create(this._context, this.partitionId, this._receiverOptions);
-    this._context.receivers[bReceiver.name] = bReceiver;
+    this._throwIfReceiverOrConnectionClosed();
+    this._batchingReceiver = BatchingReceiver.create(this._context, this.partitionId, this._receiverOptions);
+    this._context.receivers[this._batchingReceiver.name] = this._batchingReceiver;
```

Comment:
Looks like `BatchingReceiver.create()` function takes care of setting the receiver in the `_context.receivers` map. So we don't need this step

Reply:
I didn't remove this before because I was planning to do the cleanup in the separate receiver link related issue.